test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -10860,43 +10854,40 @@ def apply_config(self, config=None, validate=True, path=None):
 
         reconfig_time = int(time.time())
         try:
-            (fd, fn) = self.du.mkstemp()
+            fn = self.du.create_temp_file()
+            fd = open(fn, "w")
             for k in self._config_order:
                 if k in config:
                     if k in self._sched_config_comments:
-                        os.write(fd, "\n".join(self._sched_config_comments[k]))
-                        os.write(fd, "\n")
+                        fd.write("\n".join(self._sched_config_comments[k]))
```

Comment:
Just a suggestion (take it or leave it up to you): Instead calling write with manually adding '\n' I suggest to create list and then make use of fd.writelines(). which will also take care of proper line ending (aka '\n' for Linux and in future '\r\n' for windows.